### ISSUE TITLE
Chinese input method (ibus-rime) is not working for nyxt

### ISSUE THREAD

**Issue by wxie2017** [atlas-engineer/nyxt#866]
**Describe the bug**
1. using nyxt under guix 1.1.0
2. installed Chinese input method ibus-rime: it works for other applications, e.g. GNU Emacs
3. input Chinese here using rime: it is not possible to input Chinese

**Precise recipe to reproduce the issue**
1. using SUPER + SPACE key to change imput method to rime
2. input in the input field (search, etc)
3. only ascii is working

**Information**
- OS name+version: guix 1.1.0
- Graphics card and driver:
- Desktop environment / Window manager name+version:
- How you installed Nyxt (Guix pack, package manager, build from source):
- Information from command copy-system-information:

  If you can't run copy-system-information, please provide the following:
  - Nyxt version (from =M-x nyxt-version= or =nyxt --version=): 1.5.0-1.9440980
  - Lisp implementation/version (if built from source):
  - Kernel name+version:
  - WebKitGTK+ | QtWebEngine version:

**Output when started from a shell**

**Comment by Ambrevar** [comment]
This is a duplicate of #324, isn't it?

**Comment by xelxebar** [comment]
Maybe. That issue is specific to minibuffer input. I am unable to input CJK anywhere at all. @wxie2017 Is your ibus-rime input broken in input fields, the minibuffer, or both?

**Comment by Ambrevar** [comment]
CJK input should work in web input fields.  Are you using VI bindings?
Can you try on a vanilla configuration?

**Comment by wxie2017** [comment]
> Maybe. That issue is specific to minibuffer input. I am unable to input CJK anywhere at all. @wxie2017 Is your ibus-rime input broken in input fields, the minibuffer, or both?

No, this issue is not specific for minibuffer. The Chinese input for the other input field is not working, either.

**Comment by wxie2017** [comment]
> This is a duplicate of #324, isn't it?

No, no the same. Chinese character display in fine for me in nyxt. I just cannot input.

**Comment by wxie2017** [comment]
> CJK input should work in web input fields. Are you using VI bindings? Can you try on a vanilla configuration?

I have not done any customization. I just installed using `guix package install nyxt`.

**Comment by Ambrevar** [comment]
Can you reproduce in Epiphany?

**Comment by xelxebar** [comment]
@wxie2017 Could you share your input method configuration?

After a bit of playing around, I discovered that the following configuration (extracted from `.xprofile`) lets me input CJK into input fields:

    GTK_IM_MODULE=xim
    XMODIFIERS=@im=ibus
    ibus --daemonize --replace --xim

However, it completely breaks minibuffer input, exhibiting the exact same behaviour as issue #831.

Testing the above configuration shows that CJK input works with chromium and surf. Interestingly, surf also works when `GTK_IM_MODULE=ibus` while chromium does not.

**Comment by wxie2017** [comment]
> Can you reproduce in Epiphany?

Yes, Epiphany has the same issue.

**Comment by wxie2017** [comment]
> @wxie2017 Could you share your input method configuration?
> 
> After a bit of playing around, I discovered that the following configuration (extracted from `.xprofile`) lets me input CJK into input fields:
> 
> ```
> GTK_IM_MODULE=xim
> XMODIFIERS=@im=ibus
> ibus --daemonize --replace --xim
> ```
> 
> However, it completely breaks minibuffer input, exhibiting the exact same behaviour as issue #831. Note, also, that even with the input fields, input is a bit janky. CJK input methods work by letting users type phonetically and displaying a small suggestions window from which the user selects the correct orthography. Typically, this suggestions window appears right next to or below the input itself; however, when I tested the above configuration in nyxt, this window was appearing at the bottom left portion of the entire nyxt window (coordinate origin?).
> 
> Testing the above configuration shows that CJK input works with chromium and surf. Interestingly, surf also works when `GTK_IM_MODULE=ibus` while chromium does not.

my .xprofile

export GTK_IM_MODULE="ibus"
export QT_IM_MODULE="ibus"
export XMODIFIERS="@im=ibus"
ibus-daemon -d-x

**Comment by Ambrevar** [comment]
Thanks for the tip, I've added it to the manual in
3c1be80e18ce741df25eaa0a548d04e1f28d1d8a.

**Comment by xelxebar** [comment]
@Ambrevar Um. I wouldn't recommend those settings. As I mentioned, they make the minibuffer completely unusable. Plus, they only make CJK input possible, but it's still partially broken.

**Comment by Ambrevar** [comment]
I've cited them as an example, we need to work out better settings.

**Comment by jmercouris** [comment]
This should be working now with the new prompt-buffer no?

**Comment by Ambrevar** [comment]
According to https://github.com/atlas-engineer/nyxt/issues/866#issuecomment-664209656 this is not related to the prompt-buffer, but rather a GTK issue.

**Comment by jmercouris** [comment]
I'm saying it should be possible to use xelxebars recommended settings with the prompt buffer, oder?

**Comment by wxie2017** [comment]
Pierre Neidhardt ***@***.***> writes:

> Maybe indeed. @xelxebar @wxie2017 Can you tell us if the latest Nyxt works for you now,
> with which settings and how it compares with Epiphany?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

guix (GNU Guix) 510e24f973a918391d8122fd6ad515c0567bf23e
Nyxt version 2-pre-release-6

Using Nyxt and Epiphany to open
https://www.qwant.com

1. In the search field, neither Nyxt nor Epiphany accepts Chinese input.
2. Copy and Paste to the search field is ok for both.
3. Chinese input for GNU Emacs is fine.

Difference between Nyxt and Epiphany:

The position of input field of https://www.qwant.com is changing from
page middle to page top when focus changing to it.

It is not bad.

best regards
wxie

**Comment by Ambrevar** [comment]
Thank you for your feedback!

> Using Nyxt and Epiphany to open
> https://www.qwant.com
> > 1. In the search field, neither Nyxt nor Epiphany accepts Chinese input.
> 2. Copy and Paste to the search field is ok for both.
> 3. Chinese input for GNU Emacs is fine.

Can you try @xelxebar's suggestion?

```
GTK_IM_MODULE=xim
***@***.***=ibus
ibus --daemonize --replace --xim
```

> The position of input field of https://www.qwant.com is changing from
> page middle to page top when focus changing to it.

I can reproduce this both in Nyxt and Epiphany, when the window is
narrow enough.  I think this is just the website being responsive, I
don't think there a Nyxt.

**Comment by xelxebar** [comment]
Nice to see this issue moving forward!

Unfortunately, I am still unable to input Japanese anywhere:

- prompt-buffer: Completely unresponsive when Japanese input enabled;
- HTML text fields: input method registers keystrokes and displays suggestions, however, the text field remains empty.

Compared to Epiphany:

- Search bar: same as text input in Nyxt above
- HTML text fields: operate as expected.

```
$ guix package -I nyxt
nyxt	2-pre-release-6	out	/gnu/store/zk06ngmk9zh32acc59v7y1higbvfv3qb-nyxt-2-pre-release-6
```

**Comment by Ambrevar** [comment]
Which input system are you using?
Is it specific to Japanese, are have you tried with other languages
using input systems?

**Comment by xelxebar** [comment]
Argh. Sorry. I should have included that info from the beginning:

```
$ guix package -I ibus
ibus-anthy	1.5.9	out	/gnu/store/0rp8igxc1d709snla5iy58sm211pgmzx-ibus-anthy-1.5.9
ibus-rime    1.4.0   out     /gnu/store/i0aagm009cl9b3p26f0pssm7a98db6y8-ibus-rime-1.4.0
ibus	1.5.22	out	/gnu/store/hcv3vb9vvh7zw0zab45yd31gpmgl17iq-ibus-1.5.22
```

Both rime (Chinese) and anthy (Japanese) behave the same for me. Unfortunately, Guix hasn't yet packaged any CJK input methods other than ibus, so that's all I've been able to test so far.

**Comment by Ambrevar** [comment]
There is also UIM on Guix.

**Comment by xelxebar** [comment]
Ah! I didn't realize that Guix compiled in everything. Unfortunately, I'm unable to get UIM input working at all.
